src/audio/drc/drc.c

```diff
@@ -121,7 +124,7 @@ inline int drc_set_pre_delay_time(struct drc_state *state,
 
 static int drc_setup(struct drc_comp_data *cd, uint16_t channels, uint32_t rate)
 {
-	uint32_t sample_bytes = get_sample_bytes(cd->source_format);
+	uint32_t sample_bytes = get_sample_bytes(cd->sink_format);
```

Comment:
is this a fix in the current code? Should it be a pre-requisite separate patch then?

Reply:
In IPC4 components the sink buffer params are set with init() IPC parameters so better to get the stream information from sink instead of source. I could of course make this change a separate IPC4 changes and additions commit.

Reply:
fixed now